src/CredentialManager.cpp

```diff
@@ -0,0 +1,1040 @@
+}
+
+bool CredentialManager::removeCredentialFromFile(const QString& profileName, const QString& key)
+{
+    QString filePath = generateFilePath(profileName, key);
+    
+    // Check if path generation failed due to validation
+    if (filePath.isEmpty()) {
+        qWarning() << "CredentialManager: Failed to generate valid file path for removing encrypted credential";
+        return false;
+    }
+    
+    // Check if file exists before attempting removal
+    if (!QFile::exists(filePath)) {
+        // Not an error - credential may not have been stored or already removed
+        return true;
+    }
+    
+    bool removed = QFile::remove(filePath);
+
+    if (!removed) {
+        qWarning() << "CredentialManager: Failed to remove encrypted credential file:" << filePath;
+    }
+    
+    return removed;
+}
+
+QString CredentialManager::generateServiceName(const QString& profileName, const QString& key)
+{
+    // Sanitize inputs to prevent keychain service name conflicts
+    auto sanitizeForService = [](const QString& input) -> QString {
+        QString sanitized = input;
+        // Replace invalid characters with underscores
+        sanitized.replace(QRegularExpression(R"([^\w\-\.])"), "_");
+        // Limit length to prevent overly long service names
+        if (sanitized.length() > 50) {
+            sanitized = sanitized.left(50);
+        }
+
+        return sanitized;
+    };
+    
+    QString sanitizedProfile = sanitizeForService(profileName);
+    QString sanitizedKey = sanitizeForService(key);
+    
+    return QString("Mudlet-%1-%2").arg(sanitizedProfile, sanitizedKey);
+}
+
+QString CredentialManager::generateFilePath(const QString& profileName, const QString& key)
+{
+    // Validate and sanitize file path components
+    if (profileName.isEmpty() || key.isEmpty()) {
+        qWarning() << "CredentialManager: Empty profile name or key provided for file path";
+        return QString();
+    }
+    
+    // Check for invalid characters that could cause path traversal or filesystem issues
+    QRegularExpression pathTraversalPattern(R"(\.\.|[<>:"|?*\x00-\x1f])");
+
+    if (profileName.contains(pathTraversalPattern) || key.contains(pathTraversalPattern)) {
+        qWarning() << "CredentialManager: Invalid characters detected in path components";
+        return QString();
+    }
+    
+    auto sanitizeForPath = [](const QString& input) -> QString {
+        QString sanitized = input;
+        // Replace filesystem-unsafe characters with underscores
+        sanitized.replace(QRegularExpression(R"([/\\:*?"<>|])"), "_");
+        // Limit length to prevent filesystem issues
+        if (sanitized.length() > 50) {
+            sanitized = sanitized.left(50);
+        }
+
+        return sanitized;
+    };
+    
+    QString sanitizedProfile = sanitizeForPath(profileName);
+    QString sanitizedKey = sanitizeForPath(key);
+    
+    QString configPath = QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation);
```

Comment:
Does this take Mudlet running in portable mode into account?

Reply:
Added comments explaining how generateFilePath() handles portable mode transparently